Clarify documentation for DragAnchorStrategy.

Hixie opened the pull request:
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

Hixie committed:
Clarify documentation for DragAnchorStrategy.

Hixie commented:
cc @prosac @chunhtai

chunhtai reviewed (approved):
LGTM, except test failure

Hixie commented:
Weird, the same API in LongPressDraggable wasn't deprecated. Was that intentional?

I've updated the PR to deprecate the API there too, and updated the example accordingly.

chunhtai commented:
> Weird, the same API in LongPressDraggable wasn't deprecated. Was that intentional?
> 
> I've updated the PR to deprecate the API there too, and updated the example accordingly.

The intention is to do that in a separate pr. I wonder if @prosac has one in progress? Since you already include in this PR, I think we should just merge this

chunhtai commented on packages/flutter/lib/src/widgets/drag_target.dart:
```diff
@@ -435,7 +492,14 @@ class LongPressDraggable<T extends Object> extends Draggable<T> {
+      'This feature was deprecated after v2.1.0-10.0.pre.'
+    )
     DragAnchor dragAnchor = DragAnchor.child,
+    DragAnchorStrategy? dragAnchorStrategy,
```
We may need a test for this. Since Draggable already has test coverage, We can probably just create the LongPressDraggable with dragAnchorStrategy and make sure the same callback is passed into Draggable

chunhtai reviewed (commented):
We should have a test for LongPressDraggable

Hixie commented:
Is there more to it than I think? The change to do that was basically 5 lines as far as I can tell.

chunhtai commented:
> Is there more to it than I think? The change to do that was basically 5 lines as far as I can tell.

No I think this is it.

chunhtai commented:
Hi @Hixie are you still working on this PR?

Hixie commented:
Yeah, though feel free to take it if you want to run with it.

Hixie commented:
Added a test as suggested.

chunhtai reviewed (approved):
LGTM with the test.

prosac commented:
hi! i during the weekend i planned to implement the change for `LongPressDraggable` but ended up making the following suggestion-PR: https://github.com/flutter/flutter/pull/79784

prosac commented:
i will continue implementing the `DragAnchorStrategy` changes when I have my next after work coding spee ;-). I guess tomorrow or sunday.

prosac commented:
wow. was too tired to check out the code yesterday. all done :1st_place_medal: . thank you @Hixie !

Hixie commented:
you're welcome :-)
